### PR TITLE
Issue/cleanup agent map

### PR DESCRIPTION
**Pull request by edvgui**
# Description

Since https://github.com/inmanta/inmanta-core/pull/7449, the agent will be started when they are added to the agent map.  This issue has gone unnoticed for so long because this setting of the environment was not reset in between tests.  This PR fixes it.

This will probably require a few labs to be rebuilt.

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] Attached issue to pull request
- [ ] Changelog entry
- [ ] Type annotations are present
- [ ] Code is clear and sufficiently documented
- [ ] No (preventable) type errors (check using make mypy or make mypy-diff)
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

### CODE REVIEW

**Review by sanderr — COMMENTED** [review]
While I'm all for it in principle, I'm not convinced it's a good idea to add it now. If I understand correctly this will cause tests to start failing for all versions except the latest iso ones. Is that right? If so, I think we should wait for at least a year before we make this change.

File: src/pytest_inmanta_lsm/plugin.py [hunk 1]
```diff
@@ -433,6 +433,17 @@ def remote_orchestrator(
     }
     settings.update(remote_orchestrator_settings)
 
+    # Get server version and check if the server has the patch which safely allows to reset
+    # the agent map
+    # https://github.com/inmanta/inmanta-core/issues/7448
+    orchestrator_version = version.Version(remote_orchestrator_shared.status.version)
+    good_iso7 = orchestrator_version > version.Version("7.1.0.dev")
+    good_iso6 = version.Version("7.dev") > orchestrator_version > version.Version("6.5.0.dev")
```
Comment (sanderr) [inline src/pytest_inmanta_lsm/plugin.py:441]:
```suggestion
    good_iso6 = version.Version("7") > orchestrator_version > version.Version("6.5.0.dev")
```
`7 < 7.dev`

**Review by sanderr — CHANGES_REQUESTED** [review]

File: src/pytest_inmanta_lsm/plugin.py (hunk 1 above)
Comment (edvgui) [reply src/pytest_inmanta_lsm/plugin.py:441]:
```
>>> version.Version("7") < version.Version("7.dev")
False
```

Comment (sanderr) [reply src/pytest_inmanta_lsm/plugin.py:441]: You're absolutely right, not sure what I was thinking.

**Pull request closed without merging**